### PR TITLE
[8.4] MOD-12807, MOD-12808: Register Cursor Sub-Commands as such

### PR DESCRIPTION
**Pull request by Itzikvaknin**
Manual backport of https://github.com/RediSearch/RediSearch/pull/7571 to 8.4

---

> [!NOTE]
> **Register FT.CURSOR subcommands and refactor cursor handling**
> 
> - Split `FT.CURSOR` into true subcommands: `READ`, `PROFILE`, `DEL`, `GC`; register both external (`FT.CURSOR`) and internal (`_FT.CURSOR`) variants with cluster-safe dispatch
> - Replace monolithic `RSCursorCommand` with `RSCursorReadCommand`, `RSCursorProfileCommand`, `RSCursorDelCommand`, `RSCursorGCCommand`; adjust distributed wrappers and commandstats names
> - Change `sendChunk_ReplyOnly_EmptyResults` and cursor read paths to take `RedisModuleCtx`, internally create/end replies, and improve RESP2/RESP3 empty reply behavior and OOM warnings
> - Improve argument validation and error messages (bad/missing cursor id, unknown args/subcommands, wrong arity), and handle index-dropped cases
> - Update tests to new APIs/behaviors, including PROFILE handling, error texts, and GC edge cases
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit a6e9806c1da3b55d3e1bee6af29bc87f79cbb33e. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/aggregate/aggregate_exec.c [hunk 1]
```diff
@@ -1376,118 +1378,144 @@ static void cursorRead_ctx(CursorReadCtx *cr_ctx) {
...
+  RS_ASSERT(cursor_HasSpecWeakRef(cursor));
+  // Check if the spec is still valid
+  StrongRef execution_ref = IndexSpecRef_Promote(cursor->spec_ref);
+  if (!StrongRef_Get(execution_ref)) {
+    // The index was dropped while the cursor was idle.
+    // Notify the client that the query was aborted.
+    RedisModule_ReplyWithError(ctx, "The index was dropped while the cursor was idle");
   } else {
-    RedisModule_Reply_Error(reply, "Unknown subcommand");
+    sendChunk_ReplyOnly_EmptyResults(ctx, req);
```
[the remark this answers is not in the document]
Comment (Itzikvaknin) [reply src/aggregate/aggregate_exec.c:1462]: @meiravgri ?

File: src/aggregate/aggregate_exec.c [hunk 2]
```diff
@@ -1321,8 +1323,8 @@ static void cursorRead(RedisModule_Reply *reply, Cursor *cursor, size_t count, b
     if (!StrongRef_Get(execution_ref)) {
       // The index was dropped while the cursor was idle.
       // Notify the client that the query was aborted.
-      Cursor_Free(cursor); // Free the cursor since it is no longer valid
```
Comment (GuyAv46) [inline src/aggregate/aggregate_exec.c:1324]: Consider keeping the comment

**Review by GuyAv46 — APPROVED** [review]
